pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -2018,25 +2032,25 @@ impl Timeline {
                 }
                 accumulated_values.push((key, lsn, val));
             } else {
-                let last_key = last_key.as_mut().unwrap();
-                stat.on_unique_key_visited();
-                let skip_adding_key = if let Some(ref compaction_key_range) = compaction_key_range {
-                    !compaction_key_range.contains(last_key)
-                } else {
-                    false
-                };
+                let last_key: &mut Key = last_key.as_mut().unwrap();
+                stat.on_unique_key_visited(); // TODO: adjust statistics for partial compaction
```

Comment:
Tracking this in a follow-up task already? I'm not sure I understand what needs to be done.

Reply:
It depends on whether we want to include keys not in the partial compaction range into the statistics. Currently I use the statistics to know how much space is saved by running the compaction.